src/reflect/scala/reflect/internal/transform/Erasure.scala

```diff
@@ -112,8 +112,9 @@ trait Erasure {
     protected def eraseDerivedValueClassRef(tref: TypeRef): Type = erasedValueClassArg(tref)
 
     def apply(tp: Type): Type = tp match {
-      case ConstantType(_) =>
-        tp
+      case ConstantType(ct) =>
+        if (ct.tag == ClazzTag) ConstantType(Constant(apply(ct.typeValue)))
+        else tp
```

Comment:
i do see the difference in the AST, i wonder if that's expected and what happens in the backend so that the two fields hold the same value:
```
$ cat Test.scala
object Test {
  final val a              = classOf[Unit]
  final val b: Class[Unit] = classOf[Unit]
}
$ scalac Test.scala -Xprint:erasure
[[syntax trees at end of                   erasure]] // Test.scala
package <empty> {
  object Test extends Object {
    def <init>(): Test.type = {
      Test.super.<init>();
      ()
    };
    final private[this] val a: Class(classOf[scala.runtime.BoxedUnit]) = classOf[scala.runtime.BoxedUnit];
    final <stable> <accessor> def a(): Class(classOf[scala.runtime.BoxedUnit]) = classOf[scala.runtime.BoxedUnit];
    final private[this] val b: Class = classOf[scala.Unit];
    final <stable> <accessor> def b(): Class = Test.this.b
  }
}
```